xdsl/dialects/llvm.py

```diff
@@ -1502,6 +1502,29 @@ def parse_parameters(cls, parser: AttrParser) -> list[Attribute]:
         parser.raise_error("Unknown calling convention")
 
 
+class FramePointerKind(StrEnum):
+    NONE = "none"
+    NONLEAF = "non-leaf"
+    ALL = "all"
+    RESERVED = "reserved"
+
+
+@irdl_attr_definition
+class FramePointerKindAttr(EnumAttribute[FramePointerKind]):
+    """LLVM Frame Pointer Kind."""
+
+    name = "llvm.framePointerKind"
+
+    def print_parameter(self, printer: Printer) -> None:
+        with printer.in_angle_brackets():
+            super().print_parameter(printer)
```

Comment:
It surprises me that printing the parameters in angle brackets isn't the default behaviour

Reply:
The reason is kinda wild.

There are two ways to print an attribute. Either you use the "dialect format", which is `#dialect<enum<...>>`, or you use the "pretty format", which is `#dialect.enum<...>`.
You can only use the "pretty format" if you use the angle brackets.

So, why would you even use the "dialect format"? Well, if you have an operation that knows statically the attribute type, then you only print the `...` instead of `#dialect<enum ...>`. In particular, that means that you automatically print `all` instead of `#dialect<enum all>`. If you had angle brackets, this would be `<all>` instead of `#dialect<enum<all>>`

Reply:
(Btw, I checked and it is the default behavior of attributes in our codebase, but not enums)